From: Takashi Iwai <tiwai@suse.de>
Date: Wed, 21 Jan 2009 15:53:36 +0000 (+0100)
Subject: Fix build of pph plugin with speex-1.2rc1
X-Git-Tag: v1.0.20~10
X-Git-Url: http://git.alsa-project.org/?p=alsa-plugins.git;a=commitdiff_plain;h=ba7cd8e0a04798479f851741ec29fef97889aa39;hp=c547b535a20a0909c17cb6dd14e5e364b2ae0fe9

Fix build of pph plugin with speex-1.2rc1

The resampler code is moved to libspeexdsp instead of libspeex.

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---

diff --git a/configure.in b/configure.in
index 837934f..ce95b5d 100644
--- a/configure.in
+++ b/configure.in
@@ -106,16 +106,18 @@ AC_SUBST(AVCODEC_CFLAGS)
 AC_SUBST(AVCODEC_LIBS)
 AC_SUBST(AVCODEC_HEADER)
 
-USE_LIBSPEEX=""
+PKG_CHECK_MODULES(speexdsp, [speexdsp >= 1.2], [HAVE_SPEEXDSP="yes"], [HAVE_SPEEXDSP=""])
+AM_CONDITIONAL(HAVE_SPEEXDSP, test "$HAVE_SPEEXDSP" = "yes")
+
 AC_ARG_WITH([speex],
 	AS_HELP_STRING([--with-speex={builtin|lib|no}],
 		[build speex resampler (built-in code, link with external lib, or no build)]),
 	[PPH=$withval], [PPH="lib"])
 
+USE_LIBSPEEX=""
 if test "$PPH" = "lib"; then
-	PKG_CHECK_MODULES(speex, [speex >= 1.2], [USE_LIBSPEEX="yes"], [USE_LIBSPEEX=""])
-	if test "$USE_LIBSPEEX" = "yes"; then
-		AC_CHECK_LIB([speex], [speex_resampler_init],
+	if test "$HAVE_SPEEXDSP" = "yes"; then
+		AC_CHECK_LIB([speexdsp], [speex_resampler_init],
 			[USE_LIBSPEEX="yes"], [USE_LIBSPEEX=""])
 	fi
 	if test "$USE_LIBSPEEX" = "yes"; then
@@ -197,6 +199,6 @@ if test "$HAVE_AVCODEC" = "yes"; then
 fi
 echo "Speex rate plugin:  $PPH"
 if test "$PPH" = "lib"; then
-  echo "  speex_CFLAGS: $speex_CFLAGS"
-  echo "  speex_LIBS: $speex_LIBS"
+  echo "  speexdsp_CFLAGS: $speexdsp_CFLAGS"
+  echo "  speexdsp_LIBS: $speexdsp_LIBS"
 fi
diff --git a/pph/Makefile.am b/pph/Makefile.am
index d24176c..551e5bd 100644
--- a/pph/Makefile.am
+++ b/pph/Makefile.am
@@ -2,15 +2,16 @@ asound_module_rate_speexrate_LTLIBRARIES = libasound_module_rate_speexrate.la
 
 asound_module_rate_speexratedir = @ALSA_PLUGIN_DIR@
 
-AM_CFLAGS = -DVAR_ARRAYS -DRANDOM_PREFIX=alsa_lib -DOUTSIDE_SPEEX -Wall -g @ALSA_CFLAGS@
+AM_CFLAGS = -DVAR_ARRAYS -Wall -g @ALSA_CFLAGS@
 AM_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined $(LDFLAGS_NOUNDEFINED)
 
 libasound_module_rate_speexrate_la_SOURCES = rate_speexrate.c
 libasound_module_rate_speexrate_la_LIBADD = @ALSA_LIBS@
 if USE_LIBSPEEX
-AM_CFLAGS += @speex_CFLAGS@
-libasound_module_rate_speexrate_la_LIBADD += @speex_LIBS@
+AM_CFLAGS += @speexdsp_CFLAGS@
+libasound_module_rate_speexrate_la_LIBADD += @speexdsp_LIBS@
 else
+AM_CFLAGS += -DRANDOM_PREFIX=alsa_lib -DOUTSIDE_SPEEX 
 libasound_module_rate_speexrate_la_SOURCES += resample.c
 libasound_module_rate_speexrate_la_LIBADD += -lm
 endif
